Sources/Subs-Db-mysql.php

```diff
@@ -776,13 +776,24 @@ function smf_db_insert($method = 'replace', $table, $columns, $data, $keys, $ret
 	global $smcFunc, $db_connection, $db_prefix;
 
 	$connection = $connection === null ? $db_connection : $connection;
+	
+	$return_var;
```

Comment:
https://secure.php.net/empty

Yes, null is considered empty.  But you can also test for null with is_null or === null.  Which would make it different than 0 or false.